### PR TITLE
feat(gitopscli/*): add merge method parameter to deploy command

### PR DESCRIPTION
**Pull request by socar-humphrey**
Hi, 

In my usecase, I needed to select a merge method other than `Create a Merge Commit`, but there's no merge method selection option on deploy command by default.

I added a merge method parameter to deploy command, so that it enables users to pick one of `squash`, `rebase`, `merge commit`(if it is set to None)

### CODE REVIEW

File: gitopscli/cliparser.py [hunk 1]
```diff
@@ -99,6 +99,8 @@ def __create_deploy_parser() -> ArgumentParser:
         const=True,
         default=False,
     )
+    parser.add_argument("--merge-method", help="Merge Method (squash, rebase, merge commit if None)", type=str,
```
Comment (joachimprinzbach) [inline gitopscli/cliparser.py:102]: Maybe use "merge-commit" as default instead of none. Syntactically, "none" would not be a valid method

**Review by joachimprinzbach — COMMENTED** [review]
Would be great to have the docs updated as well: https://github.com/baloise/gitopscli/blob/master/docs/commands/deploy.md

File: gitopscli/cliparser.py [hunk 2]
```diff
@@ -99,6 +99,8 @@ def __create_deploy_parser() -> ArgumentParser:
         const=True,
         default=False,
     )
+    parser.add_argument("--merge-method", help="Merge Method (e.g., 'squash', 'rebase', 'merge') (default: merge)", type=str,
+                        default="merge")
```
Comment (christiansiegel) [inline gitopscli/cliparser.py:103]: If you default to `merge` then defining `merge_method` as `Optional[..]` makes no sense.

However, I would remove the default value from the cli. Hence, if the `--merge-method` is not specified, the Git server can use its default merge method -> no breaking change for existing users.

File: gitopscli/git_api/github_git_repo_api_adapter.py [hunk 3]
```diff
@@ -36,9 +37,9 @@ def create_pull_request(
         pull_request = repo.create_pull(title=title, body=description, head=from_branch, base=to_branch)
         return GitRepoApi.PullRequestIdAndUrl(pr_id=pull_request.number, url=pull_request.html_url)
 
-    def merge_pull_request(self, pr_id: int) -> None:
+    def merge_pull_request(self, pr_id: int, merge_method: Optional[Literal["squash", "rebase", "merge"]] = "merge") -> None:
```
Comment (christiansiegel) [inline gitopscli/git_api/github_git_repo_api_adapter.py:40]: You did not implement the `merge_method` parameter for the Bitbucket and GitLab implementations.

**Review by christiansiegel — CHANGES_REQUESTED** [review]
I don't know why travis is not building 🤔 

There are some broken tests, formatting, etc. I added a Readme section on how to run those in #156

**Comment by socar-humphrey** [comment]
A little tip for better DX: add [pre-commit](https://travis-ci.com/github/baloise/gitopscli/jobs/520716793) for formatting and tests.
https://travis-ci.com/github/baloise/gitopscli/jobs/520716793

This build has failed bc of a little formatting issue. Pre-commit enables codes not only to be in the project-wide format setting in commit level, but also to be passing tests before making a commit. It did boost my team's development cycle IMO.

**Comment by christiansiegel** [comment]
> A little tip for better DX: add [pre-commit](https://travis-ci.com/github/baloise/gitopscli/jobs/520716793) for formatting and tests.
> https://travis-ci.com/github/baloise/gitopscli/jobs/520716793

Good idea! Added in #158

**Comment by socar-humphrey** [comment]
@christiansiegel @joachimprinzbach 
Plz lmk if I need to add anything else for this PR. (I think its ready for merge)

**Comment by joachimprinzbach** [comment]
lgtm - @christiansiegel could you approve pls?

**Review by christiansiegel — APPROVED** [review]
I'm still missing any doc or log message for Bitbucket and Gitlab if the chosen merge method is not supported and just ignored. But we can add that later..

Thanks for the feature @socar-humphrey!

**Pull request merged**